File: Week4/problemA.cpp
/*
Language: cpp
ID: 61975
QLink: https://quera.org/course/assignments/61975/problems
Author: @Amirkhaskar
*/

#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

int numRabbits(vector<int> &answers)
{
    unordered_map<int, int> rabbit;
    int sum = 0;
    for (int i = 0; i < answers.size(); i++)
    {
        if (rabbit[answers[i]] == 0)
        {
            sum += (answers[i] + 1);
            rabbit[answers[i]] = answers[i];
        }
        else
        {
            rabbit[answers[i]]--;
        }
    }
    return sum;
}

int main()
{
    vector<int> x;
    int num;
    while (cin >> num)
    {
        x.push_back(num);
    }
    cout << numRabbits(x) << endl;
    return 0;
}

File: Week4/problemA.cs
/*
Language: csharp
ID: 61975
QLink: https://quera.org/course/assignments/61975/problems
Author: @Amirkhaskar
*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;


public class Program
{
    public static void Main()
    {
        String answers = Console.ReadLine();
        int sum = 0;
        int[] tmp = new int[1000];

        String[] spearator = { " " };
        String[] answersArray = answers.Split(spearator, 3,
               StringSplitOptions.RemoveEmptyEntries);

        for (int i = 0; i < (answersArray.Length); i++)
        {

            int rab = int.Parse(answersArray[i]);
            if (tmp[rab] > 0)
            {
                tmp[rab]--;
            }
            else
            {
                tmp[rab] = rab;
                sum += rab + 1;
            }

        }

        Console.WriteLine(sum);
    }
}



